datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -52,20 +78,57 @@
    *                           {@value DEFAULT_IMBALANCE_THRESHOLD}.
    * @param maxPartitionPerTask The maximum number of partitions allowed per task. By default it's Integer.MAX (no limit)
    *                            If partitions count in task is larger than this number, Brooklin will throw an exception
+   * @param enableElasticTaskAssignment A boolean indicating whether elastic task assignment is enabled or not. If
+   *                                    enabled, the number of tasks to assign will be determined by a minTasks
+   *                                    datastream metadata property, and may be increased based on the number of tasks
+   *                                    determined during partition assignment.
+   * @param partitionsPerTask If elastic task assignment is enabled, this is used to determine the number of partitions
+   *                          allowed in each task when determining the number of tasks for the first time. The default
+   *                          is {@value DEFAULT_PARTITIONS_PER_TASK}.
+   * @param partitionFullnessFactorPct If elastic task assignment is enabled, this is used to determine how full to
+   *                                   the tasks when fitting partitions into them for the first time. The default
+   *                                   is {@value DEFAULT_PARTITION_FULLNESS_FACTOR_PCT}.
    *
    */
   public StickyPartitionAssignmentStrategy(Optional<Integer> maxTasks, Optional<Integer> imbalanceThreshold,
-      Optional<Integer> maxPartitionPerTask) {
+      Optional<Integer> maxPartitionPerTask, boolean enableElasticTaskAssignment, Optional<Integer> partitionsPerTask,
+      Optional<Integer> partitionFullnessFactorPct) {
     super(maxTasks, imbalanceThreshold);
+    _enableElasticTaskAssignment = enableElasticTaskAssignment;
     _maxPartitionPerTask = maxPartitionPerTask.orElse(Integer.MAX_VALUE);
+    _partitionsPerTask = partitionsPerTask.orElse(DEFAULT_PARTITIONS_PER_TASK);
+    _partitionFullnessFactorPct = partitionFullnessFactorPct.orElse(DEFAULT_PARTITION_FULLNESS_FACTOR_PCT);
+
+    LOG.info("Elastic task assignment is {}", _enableElasticTaskAssignment ? "enabled" : "disabled");
```

Comment:
Please add other inputs as well in the log.

Reply:
done